src/common/snippets/src/pass/mlp_seq_tokenization.cpp

```diff
@@ -0,0 +1,207 @@
+        size_t potential_body_params_count = 2;
+        // After some transformations, a different number of Constants for some operations may be created
+        // than the actual number of Constants during tokenization.
+        // To avoid unsupported number of non-scalar Constants in the future (plugin specific limitation)
+        // we should calculate potential number of non-scalar Constants that will be moved up from body.
+        // Heuiristic count of possible buffers - upper-bound value
+        const size_t uniqie_buffer_reg_group_count = 2;
+        std::string fused_names;
+        ov::NodeVector ordered_ops;
+
+        /* ======== Matcher Pass ========== */
+
+        /****** Skeleton ******/
+        /* Skeleton on MLP sequence-pattern is:
+         *            MatMul0
+         *               |
+         *    Eltwise/FakeQuantize (1 or more)
+         *               |
+         *            MatMul1
+         *               |
+         *    Eltwise/FakeQuantize (1 or more)
+         *               |
+         *            MatMul2
+         *               |
+         *              ...
+         */
+
+        const auto matmul0 = ov::as_type_ptr<ov::opset1::MatMul>(pattern_to_output.at(m_matmul0).get_node_shared_ptr());
+        if (!is_matmul_supported(matmul0) || !has_one_consumer(matmul0)) {
+            return false;
+        }
+
+        if (transformation_callback(matmul0)) {
+            return false;
+        }
+
+        // Add possible FQ before matmul0
+        if (const auto fq = ov::as_type_ptr<ov::op::v0::FakeQuantize>(matmul0->input_value(0).get_node_shared_ptr())) {
+            if (has_one_consumer(fq)) {
+                potential_body_params_count += ov::snippets::utils::get_non_scalar_constant_count_for_fq(fq);
+                ordered_ops.push_back(fq);
+            }
+        }
+        ordered_ops.push_back(matmul0);
+
+        // Tokenize Sequence while we can do it
+        std::shared_ptr<ov::Node> prev_op = matmul0;
+        auto interm_op = prev_op->get_output_target_inputs(0).begin()->get_node()->shared_from_this();
+        auto available_regs = config.get_data_ptr_gpr_count();
+
+        while (has_one_consumer(prev_op)) {
+            auto possible_param_count = potential_body_params_count;
+            auto possible_potential_body_params_count = potential_body_params_count;
+
+            if (is_matmul_supported(interm_op) && !transformation_callback(interm_op)) {
+                // +1 for weights
+                possible_param_count++;
+            } else if (is_supported_intermediate_op(interm_op)) {
+                possible_param_count += get_potential_body_params(interm_op);
+                if (const auto fq = ov::as_type_ptr<ov::op::v0::FakeQuantize>(matmul0->input_value(0).get_node_shared_ptr())) {
+                    possible_potential_body_params_count +=
+                        ov::snippets::utils::get_non_scalar_constant_count_for_fq(fq);
+                }
```

Comment:
Let's move FQ specific logic to `get_potential_body_params` - we anyway handle FQ in a special way there

Reply:
I'm not sure how, because `get_potential_body_params` is performed on `interm_op`, but here we have `matmul0` input

Reply:
But it must be performed on `interm_op` -- FQ on 0th input has already been checked on L137-142. Here we try to tokenize ops after matmul, so please change a parameter on `interm_op`